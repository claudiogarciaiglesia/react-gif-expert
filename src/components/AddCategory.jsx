import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onFormSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length > 1) {
            // setCategories((categories) => [...categories, inputValue])
            onNewCategory(inputValue.trim())
            setInputValue('')
        }

    }

    return (
        <form onSubmit={onFormSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Search for gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}