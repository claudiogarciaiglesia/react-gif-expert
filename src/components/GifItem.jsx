export const GifItem = ({ title, url }) => {
    return (
        <div className="card">
            <img className="card-img" src={url} alt={title} />
            <p>{title}</p>
        </div>
    )

}
