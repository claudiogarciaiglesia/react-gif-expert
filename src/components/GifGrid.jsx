import { useEffect, useState } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category)
    return (
        <>
            <h3 aria-label="gif-grid">{category}</h3>
            {isLoading && (<h3>Cargando...</h3>)}

            <div className="card-grid">
                {images.map((image) => (
                    <GifItem
                        key={image.id}
                        {...image}
                    />
                ))}
            </div >
        </>)
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}