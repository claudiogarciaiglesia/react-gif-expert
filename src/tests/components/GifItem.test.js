import { render, screen } from "@testing-library/react"
import { GifItem } from "../../components/GifItem"

describe('<GifItem /> testing', () => {

    const title = 'title test';
    const category = 'category';
    const url = `https://api.giphy.com/v1/gifs/search?api_key=QzBgef9jsgl7F0xzkx5S1e3g6mys77Bk&q=${category}&limit=10`;

    test('should match snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();

    });

    test('should display img with url and specified alt', () => {

        render(<GifItem title={title} url={url} />);

        const { src, alt } = screen.getByRole('img');

        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        expect(src).toBe(url);
        expect(alt).toBe(title);

    })

    test('should display component title', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })

})