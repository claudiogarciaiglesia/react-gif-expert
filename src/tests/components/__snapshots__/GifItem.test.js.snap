// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> testing should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title test"
      class="card-img"
      src="https://api.giphy.com/v1/gifs/search?api_key=QzBgef9jsgl7F0xzkx5S1e3g6mys77Bk&q=category&limit=10"
    />
    <p>
      title test
    </p>
  </div>
</div>
`;
